### PR TITLE
For Custom fields (those with no type), allow defaults to be set

### PR DESCRIPTION
**Pull request by barrymay**
For custom fields, the register was returning prior to the default being allowed to be set

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -228,42 +228,46 @@ export default function useForm<Data extends DataType>(
...
+            validate,
+            ref: { type: 'radio', name },
+          };
+        if (validate) fields[name]!.validate = validate;
+
+        (fields[name]!.options || []).push({
+          ...inputData,
+          mutationWatcher: onDomRemove(
+            elementRef,
+            (): Function => removeEventListener(inputData, true),
```
Comment (bluebill1049) [inline src/index.ts]: This may need to be avoided without type.

**Comment by bluebill1049** [comment]
curious to know what's the reason for this change?

File: src/index.ts (hunk 1 above)
Comment (barry4dev) [reply src/index.ts]: this diff is just largely visual - this change largely just moves the return for !type till after the isRadio check - I'll explain more on your other question

**Comment by barry4dev** [comment]
Reason for the change -

If I do this
```ts
useForm<FormData>({
    mode: 'onSubmit',
    defaultValues: {
      foo: [],
    },
  });
```

then I create a custom field
```ts
  register({name: 'foo'});
```

the foo field doesn't get the empty array because the register can't get that far

**Comment by bluebill1049** [comment]
I see @barry4dev, good point :)

probably just need to omit the mutation observer bit cause it's not going to be a DOM element.

`mutationWatcher: xxx`

**Comment by barry4dev** [comment]
I don't think it's going to get there - that all is in an else clause that gets triggered only if type is set

**Comment by bluebill1049** [comment]
thanks, @barry4dev again 👍 hope you had a nice weekend

**Pull request merged**